.github/workflows/lint.yaml

```diff
@@ -7,9 +7,45 @@ on:
     types: [milestoned, opened, synchronize]
   # This allows other repositories to call this workflow in a reusable way
   workflow_call:
+    inputs:
+      udsCliVersion:
+        description: The uds-cli version to install
+        # renovate: datasource=github-tags depName=defenseunicorns/uds-cli versioning=semver
+        default: 0.16.0
+        type: string
+      udsCommonVersion:
+        description: The uds-cli version to install
+        # release-please: start-version
+        default: 0.13.1
+        # release-please: end-version
+        type: string
+
+env:
+  udsCliVersion: 0.16.0
+
 
 jobs:
+  set-uds-version:
+    runs-on: ubuntu-latest
+    outputs:
+      udsCliVersion: ${{ steps.set-uds-version.outputs.udsCliVersion }}
+      udsCommonVersion: ${{ steps.set-uds-common-version.outputs.udsCommonVersion }}
+    steps:
+      - name: Set uds version based on
+        id: set-uds-version
+        run: |
+            # If the input is not defined (i.e., on push), use the default value
+            input="${{ inputs.udsCliVersion || env.udsCliVersion }}"
+            echo "udsCliVersion=$input" >> $GITHUB_OUTPUT
+      - name: Set uds common version
+        id: set-uds-common-version
+        run: |
+            # If the input is not defined (i.e., on push), use the default value
+            input="${{ inputs.udsCommonVersion || github.head_ref }}"
+            echo "udsCommonVersion=$input" >> $GITHUB_OUTPUT
```

Comment:
this is like this because lint was set up as both a callable and on.pull_request repo. Input only works on workflow_call. Otherwise believe me i would not have done this 😅